api/types.proto

```diff
@@ -713,3 +713,34 @@ message ManagerStatus {
 	// Reachability specifies whether this node is reachable.
 	RaftMemberStatus.Reachability reachability = 4;
 }
+
+// SecretType provides information about what kind of secret this is
+enum SecretType {
+	// CONTAINER secrets are user-specified secrets that should be injected
+	// into the container as an environment variable or file.
+	CONTAINER = 0 [(gogoproto.enumvalue_customname) = "ContainerSecret"];
```

Comment:
> I don't agree that we need to flesh this out now. We have a Type exactly so we can do different things in the future.

Is this because we want to hit a date or because it doesn't make sense?

If we don't hit the registry auth use case with secrets, then the usefulness of the system is in question. I don't see how we can be putting work into security without resolving these long-standing issues.

I'm sorry to be so annoying here, but I can't really LGTM this until we hit that use case.